### PR TITLE
Prevent flash of dropdown menu on labels list

### PR DESCRIPTION
**Pull request by silverwind**
On the labels list, This `left` class caused the dropdown content to flash on page load until JS had hidden it. Remove it as I see no purpose to it.

image

### CODE REVIEW

**Review by delvh — APPROVED** [review]

**Review by yardenshoham — APPROVED** [review]

**Comment by wxiaoguang** [comment]
IIRC, it is a must, otherwise the menu overflows on a wrong direction when the screen width is small.

Could you check the git log?

**Comment by silverwind** [comment]
Looks fine on small screen:

Screenshot 2024-03-31 at 15 35 37

**Comment by silverwind** [comment]
Only issue is that if you open the menu on desktop, then downsize, it will stay right-aligned and be partially outside the screen.

**Comment by silverwind** [comment]
Okay with these new classes, it stays hidden on page load and always opens left, regardless of screen size.

**Comment by wxiaoguang** [comment]
Did a quick test, it seems that `class=menu` works well. Maybe cause the "transition" module has been removed, so the overflow bug also disappears.

So I think it is ok to only keep `class=menu` now.

**Comment by silverwind** [comment]
The problem is `.ui.menu:not(.vertical) .left.menu {display: flex}` has more specificity than `.ui.dropdown.menu {display:none}`. I don't think this usage with just `left` is something that fomantic supports.

File: templates/repo/issue/labels/label_list.tmpl [hunk 1]
```diff
@@ -8,7 +8,7 @@
 					{{ctx.Locale.Tr "repo.issues.filter_sort"}}
 				</span>
 				{{svg "octicon-triangle-down" 14 "dropdown icon"}}
-				<div class="left menu">
+				<div class="menu left transition hidden">
```
Comment (wxiaoguang) [inline templates/repo/issue/labels/label_list.tmpl]:
```suggestion
				<div class="menu">
```

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> I don't think this usage with just `left` is something that fomantic supports.

IIRC it is used by Fomantic UI JS code to calculate the popup position.

**Comment by silverwind** [comment]
The `left` class is meant only for child menus, not the main menu, so it was an abuse of fomantic:

https://fomantic-ui.com/modules/dropdown.html#menu-direction

Even thought I do prefer it to open to left, let's remove it for the sake of easier future migration off of it.

**Comment by silverwind** [comment]
What's interesting is the issue filter dropdown also open to left and don't flash, so there must be a difference.

image

**Comment by silverwind** [comment]
A I see this is a CSS override `.repository .filter.menu .ui.dropdown .menu {right: 0}` that makes them open to left.

**Review by wxiaoguang — DISMISSED** [review]
sorry it doesn't seem right by introducing these `tw-` helpers.

Fomantic UI JS just calculate the position automatically. For example, when the screen width is small, the first popup menu should be left-aligned.

**Comment by silverwind** [comment]
Added new CSS solution that will work in all these right-aligned headers. It looks better to always left-align in these imho.

**Pull request merged**